ext/pdo/pdo_sql_parser.re

```diff
@@ -315,8 +341,13 @@ rewrite:
 				memcpy(newbuffer, ptr, t);
 				newbuffer += t;
 			}
-			memcpy(newbuffer, plc->quoted, plc->qlen);
-			newbuffer += plc->qlen;
+			if (plc->quoted) {
```

Comment:
Why is this change necessary?

Reply:
AFAICT `qlen` was always `>= len`. Which isn't true for escaped question marks.

Reply:
I will check ;-)

Reply:
Without, it doesn't work properly. When running the new pdo_pgsql tests, there is a code path that sends a non-quoted "?". I'll try to figure out how and when this afternoon.